docs: revised offscreen rendering page (#26542)
electron/electron: docs/tutorial/offscreen-rendering.md

# Offscreen Rendering

## Overview

> Offscreen rendering lets you obtain the content of a browser window in a
> bitmap, so it can be rendered anywhere, for example, on texture in a 3D scene.
> The offscreen rendering in Electron uses a similar approach than the [Chromium
> Embedded Framework](https://bitbucket.org/chromiumembedded/cef) project.

Reviewer: ```suggestion
Offscreen rendering lets you obtain the content of a `BrowserWindow` in a
bitmap, so it can be rendered anywhere, for example, on texture in a 3D scene.
The offscreen rendering in Electron uses a similar approach to that of the [Chromium
Embedded Framework](https://bitbucket.org/chromiumembedded/cef) project.
```

There are two rendering modes that can be used (see the section below) and only
the dirty area is passed to the `paint` event to be more efficient. You can
stop/continue the rendering as well as set the frame rate. The specified frame
rate is a top limit value, when nothing happening on a webpage, no frames are
generated. The maximum frame rate is 60 because greater values bring only
performance losses with no benefits.

An offscreen window is always created as a

> [Frameless Window](../api/frameless-window.md).

Reviewer: Can we turn these sentences into notes rather than paragraphs?

Something like

**Notes**:
* Note 1
* Note 2

### Rendering Modes

#### GPU accelerated

GPU accelerated rendering means that the GPU is used for composition. Because of

> that, the frame has to be copied from the GPU which requires more performance,

Reviewer: ```suggestion
that, the frame has to be copied from the GPU which requires more resources,
```

thus this mode is slower than the Software output device.

> Benefits: support of WebGL and 3D CSS animations.

Reviewer: I'd revert here -- we're not using the "Benefits:` format elsewhere in the doc.

#### Software output device

This mode uses a software output device for rendering in the CPU, so the frame
generation is much faster. As a result, this mode is preferred over the GPU
accelerated one.

To enable this mode, GPU acceleration has to be disabled by calling the
[`app.disableHardwareAcceleration()`][disablehardwareacceleration] API.

## Example

Starting with a working application from the
[Quick Start Guide](quick-start.md), add the following lines to the
`main.js` file:

```javascript
const { app, BrowserWindow } = require('electron')
const fs = require('fs')

app.disableHardwareAcceleration()

let win

app.whenReady().then(() => {
  win = new BrowserWindow({ webPreferences: { offscreen: true } })

>   win.loadURL('http://github.com')

Reviewer: Can we load `https://` as a best practice?

# Offscreen Rendering

## Overview

Offscreen rendering lets you obtain the content of a `BrowserWindow` in a
bitmap, so it can be rendered anywhere, for example, on texture in a 3D scene.
The offscreen rendering in Electron uses a similar approach to that of the
[Chromium Embedded Framework](https://bitbucket.org/chromiumembedded/cef)
project.

*Notes*:

* There are two rendering modes that can be used (see the section below) and only
the dirty area is passed to the `paint` event to be more efficient.
* You can stop/continue the rendering as well as set the frame rate.
* The maximum frame rate is 60 because greater values bring only performance
losses with no benefits.

> * When nothing is happening on a webpage, no frames are generated. 

Reviewer 2: This line has a trailing whitespaces causing lint to fail.

# Offscreen Rendering

## Overview

Offscreen rendering lets you obtain the content of a `BrowserWindow` in a
bitmap, so it can be rendered anywhere, for example, on texture in a 3D scene.
The offscreen rendering in Electron uses a similar approach to that of the
[Chromium Embedded Framework](https://bitbucket.org/chromiumembedded/cef)
project.

*Notes*:

> * There are two rendering modes that can be used (see the section below) and only
> the dirty area is passed to the `paint` event to be more efficient. You can
> stop/continue the rendering as well as set the frame rate.

Reviewer: ```suggestion
* There are two rendering modes that can be used (see the section below) and only
the dirty area is passed to the `paint` event to be more efficient.
* You can stop/continue the rendering as well as set the frame rate.
```

>   The specified frame rate is a top limit value, when nothing happening on a
> webpage, no frames are generated. The maximum frame rate is 60 because greater
> values bring only performance losses with no benefits.

Reviewer: ```suggestion
  * The maximum frame rate is 60 because greater values bring only performance losses with no benefits.
  * When nothing is happening on a webpage, no frames are generated. 
```
